### ISSUE TITLE
Linux support for synchronize dark mode with OS's settings option

### ISSUE THREAD

**Issue by nordurljosahvida** [getferdi/ferdi#833]
**Is your feature request related to a problem? Please describe.**

Dark reader is an amazing addition to Ferdi, however some services such as [missive](https://missiveapp.com) support W3C `prefers-color-scheme` directive whereby the website is able to determine the system color theme [light/dark]. It even responds to automatically switching themes [my installation of ZorinOS, based on Ubuntu 18, switches between light and dark themes based on day / night time of day] and the logged-in missive website responds to such change, which is amazing.

To sum up, dark reader is perfect for websites not offering a dark theme, but for those which do, and allow reading system color theme, we should let them do so.

**Describe the solution you'd like**

Allow / enable ferdi app to read system color theme via W3C `prefers-color-theme` directive.
Currently, this doesn't work.

**Describe alternatives you've considered**

This would be the best way imo.

Thanks for the amazing work.

**Comment by nordurljosahvida** [comment]
update: as already implemented in mac: https://github.com/getferdi/ferdi/pull/173 and windows: https://github.com/getferdi/ferdi/pull/548

**Comment by vantezzen** [comment]
We use Electron's [nativeTheme component](https://www.electronjs.org/docs/api/native-theme#nativethemeshouldusedarkcolors-readonly) to check if dark mode should be enabled. It looks like this option is now also available for Linux - I'll prepare a PR for this.

**Comment by nordurljosahvida** [comment]
Thank you so much! To make sure we're talking about the same thing, does this *trigger* darkreader when system theme switches to dark, or does this let the website read the state and fraw its own conclusion regarding color theme?

To further clarify, this is the option in missive app [which is supported on ferdi via external recipe btw] automatically changes appearance based on OS theme:

![Skjámynd frá 2020-06-19 21-50-18](https://user-images.githubusercontent.com/1408210/85175046-f1c00600-b276-11ea-97cf-2f136e5dc5ff.png)

The `auto` option in the above picture works in firefox on my linux-based install but not in Ferdi on the same machine.

The ideal option would be to let website aka ferdi app make its own decisions, as native, website-provided dark theme is usually better than dark reader's "superimposed" dark theme, such as in the missive case.

Thanks!

**Comment by vantezzen** [comment]
> trigger darkreader when system theme switches to dark

This depends on the service you are using. With some services, Ferdi knows that the services can handle dark mode itself, some services have a custom darkmode.css which will get injected and the rest will use DarkReader. If you don't like how DarkReader changes your services, you can also disable DarkReader via the "Enable universal dark mode" in Ferdi's settings or disabling dark mode in Ferdi's settings for the individual services.

**Comment by nordurljosahvida** [comment]
ok thanks, i'll wait for this to show up for me and check to see how it works, thanks again!

**Comment by xerus2000** [comment]
Not stale :)

**Comment by nordurljosahvida** [comment]
Hi, is this truly fixed? I haven't seen any change in recent updates.

**Comment by usersaurus** [comment]
No, it's not fixed yet

**Comment by nordurljosahvida** [comment]
ok thanks!
